x-pack/winlogbeat/module/security/ingest/security.yml

```diff
@@ -2127,17 +2143,31 @@ processors:
           }
         }
         if (ctx?.winlog?.event_data?.AccessMask != null) {
-          ArrayList results = new ArrayList();
-          Long accessMask = Long.decode(ctx.winlog.event_data.AccessMask);
+          ArrayList list = new ArrayList();
+          long accessMask;
+          for (elem in split(ctx.winlog.event_data.AccessMask)) {
+            if (elem.length() == 0) {
+              continue;
+            }
+            list.add(elem);
+            def code = elem.replace("%%","").trim();
+            accessMask |= Long.decode(code).longValue();
+          }
+          if (list.length > 0) {
+            ctx.winlog.event_data.put("AccessMask", list);
+          }
+
+          ArrayList desc = new ArrayList();
           for (entry in params.AccessMaskDescriptions.entrySet()) {
```

Comment:
The fix would be to iterate over the word size of the flags, e.g. for the `AccessMaskDescription` case
```
for (def b = 0; b < 32; b++) {
    long accessFlag = 1 << b;
    if (accessMask&accessFlag == accessFlag) {
```
and an appropriate access to the parameter map — maybe via a hex string format of the accessFlag if an integer map key is not possible.

The equivalent change can be made for `TicketOptionsDescription` at line 876 in this file.